### PR TITLE
MP/ Add detailed diagnose screen

### PR DESCRIPTION
**Pull request by MPucheta**
**I put this as a draft because MyDiagnoses is not merged yet**

- Added detailed diagnose screen

- Added some more mixins that turned out to be repeated throughout the code

- These changes take into considerations the possibility of more answers and more than one image.

- Added pluralize

**UPDATED SCREENCAST**
preview

**SUPPORT FOR MORE THAN ONE IMAGE**

image

### CODE REVIEW

File: src/components/Text/index.js [hunk 1]
```diff
@@ -51,7 +52,7 @@ const styles = StyleSheet.create({
 Text.propTypes = {
   style: PropTypes.object,
   colorVariant: PropTypes.oneOf(['primary', 'secondary', 'black', 'white', 'gray', 'gray2', 'gray3']),
-  fontVariant: PropTypes.oneOf(['h1', 'h2', 'h3', 'description', 'body', 'date'])
+  fontVariant: PropTypes.oneOf(['h1', 'h2', 'h3', 'title', 'description', 'body', 'date'])
 }
```
Comment (nahudalla) [inline src/components/Text/index.js]: Be careful, this component is growing too much. Maybe what we need here are different components that _extend_ `Text` (by composition). Each new component would only apply its specific styles.

File: src/mixins/date/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import moment from 'moment'
+
+export const firebaseTimestampToDate = (timestamp) => (
+  moment(timestamp.seconds * 1000)
+)
```
Comment (nahudalla) [inline src/mixins/date/index.js]: This returns a `Moment` object, not a `Date` object. Consider making that clear on the function name.

File: src/mixins/text/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export const addPluralizationIf = (condition) => condition ? 's' : ''
```
Comment (nahudalla) [inline src/mixins/text/index.js]: Is there any library that does this automatically? I know of this one, but I don't know if it works for spanish: ://github.com/blakeembrey/pluralize

File: src/screens/MainApp/DetailedDiagnose/Carousel/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+      await Promise.all(
+        images.map(async (imageUID, index) => {
+          const url = await getURL(imageUID)
+          urls.push(url)
+        })
+      )
+
+      setImageExternalURLs(urls)
+      setFetchingURLs(false)
```
Comment (nahudalla) [inline src/screens/MainApp/DetailedDiagnose/Carousel/index.js]: If something fails, this statement will not execute and the state won't change to `fetchingURLs=false`. Consider wrapping the `await Promise.all` in a `try/catch/finally` block, and put this inside `finally`.

File: src/screens/MainApp/DetailedDiagnose/index.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  const date = firebaseTimestampToDate(diagnose.createdAt)
+  const answers = []
+
+  if (diagnose.answer) {
+    answers.push(
+      {
+        answer: diagnose.answer,
+        by: diagnose.answeredBy
+      }
+    )
```
Comment (nahudalla) [inline src/screens/MainApp/DetailedDiagnose/index.js]: Please place the curly brackets in the same line of the parenthesis:
```javascript
answers.push({
  answer: diagnose.answer,
  by: diagnose.answeredBy
})
```

File: src/screens/MainApp/DetailedDiagnose/index.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        <Datetime date={date} />
+        <Description description={diagnose.text} />
+        <View>
+          <Text fontVariant='title' colorVariant='black'>
+            {answers.length} Respuesta{addPluralizationIf(answers.length !== 1)}
+          </Text>
+          <Text />
+        </View>
+        <AnswerList data={answers} />
+        {answers.length === 0 && <NoAnswersYet />}
```
Comment (nahudalla) [inline src/screens/MainApp/DetailedDiagnose/index.js]: Use `isEmpty` from `lodash` here.

**Review by nahudalla — COMMENTED** [review]
Please let me know what you think about these comments.

File: src/screens/MainApp/DetailedDiagnose/Carousel/index.js (hunk 4 above)
Comment (MPucheta) [reply src/screens/MainApp/DetailedDiagnose/Carousel/index.js]: `getURL` returns the real URL, but if it fails, returns null (the carousel then shows a placeholder image).

I do not think there is something else that could fail but I'l will add the `try/catch` for safety

File: src/mixins/text/index.js (hunk 3 above)
Comment (MPucheta) [reply src/mixins/text/index.js]: I can give it a try. There is a pluralize-es but it has not been updated for 4 years now....

Comment (abelosorio) [reply src/mixins/text/index.js]: FYI I've been using this library https://www.npmjs.com/package/pluralize and it's been good so far.

Comment (MPucheta) [reply src/mixins/text/index.js]: It worked for 'respuesta', I suspect there could be some corner cases but all good for now.

**Review by nahudalla — APPROVED** [review]

**Pull request merged**